django_mongodb/query.py

```diff
@@ -0,0 +1,372 @@
+
+        `ordering` is a list with (field, ascending) tuples or a boolean --
+        use natural ordering, if any, when the argument is True and its reverse
+        otherwise.
+        """
+        if isinstance(ordering, bool):
+            # No need to add {$natural: ASCENDING} as it's the default.
+            if not ordering:
+                self.ordering.append(("$natural", DESCENDING))
+        else:
+            for field, ascending in ordering:
+                direction = ASCENDING if ascending else DESCENDING
+                self.ordering.append((field.column, direction))
+
+    @safe_call
+    def delete(self):
+        """Execute a delete query."""
+        options = self.connection.operation_flags.get("delete", {})
+        return self.collection.delete_many(self.mongo_query, **options).deleted_count
+
+    def get_cursor(self):
+        if self.query.low_mark == self.query.high_mark:
+            return []
+
+        fields = [col.target.column for col in self.columns] if self.columns else None
+        cursor = self.collection.find(self.mongo_query, fields)
+        if self.ordering:
+            cursor.sort(self.ordering)
+        if self.query.low_mark > 0:
+            cursor.skip(self.query.low_mark)
+        if self.query.high_mark is not None:
+            cursor.limit(int(self.query.high_mark - self.query.low_mark))
+        return cursor
+
+    def add_filters(self, filters, query=None):
+        children = self._get_children(filters.children)
+
+        if query is None:
+            query = self.mongo_query
+
+        if filters.connector == OR:
+            assert "$or" not in query, "Multiple ORs are not supported."
+            or_conditions = query["$or"] = []
+
+        if filters.negated:
+            self._negated = not self._negated
+
+        for child in children:
+            subquery = {} if filters.connector == OR else query
+
+            if isinstance(child, Node):
+                if filters.connector == OR and child.connector == OR and len(child.children) > 1:
+                    raise DatabaseError("Nested ORs are not supported.")
+
+                if filters.connector == OR and filters.negated:
+                    raise NotImplementedError("Negated ORs are not supported.")
+
+                self.add_filters(child, query=subquery)
+
+                if filters.connector == OR and subquery:
+                    or_conditions.extend(subquery.pop("$or", []))
+                    if subquery:
+                        or_conditions.append(subquery)
+
+                continue
+
+            try:
+                field, lookup_type, value = self._decode_child(child)
+            except FullResultSet:
+                continue
+
+            if lookup_type in ("month", "day"):
+                raise DatabaseError("MongoDB does not support month/day queries.")
+            if self._negated and lookup_type == "range":
+                raise DatabaseError("Negated range lookups are not supported.")
```

Comment:
month/day is dead code. In modern versions of Django, an error is raised elsewhere for these lookups (and will be fixed in https://github.com/mongodb-labs/django-mongodb/issues/9).

And I'm not sure why [the restriction about range queries was added](https://github.com/django-nonrel/mongodb-engine/commit/e286cfb1bc6359df69823c865245d0e9f7ab32e3). Perhaps MongoDB has added support since 2011.
`IntegerModel.objects.exclude(integer__range=(2, 4))` translates to `{'integer': {'$not': {'$gte': 2, '$lte': 4}}}` which seems to work just fine.